keps/1618-optional-gc-of-workloads/README.md

```diff
@@ -0,0 +1,428 @@
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+The following new unit tests or modifications to existing ones were added to test 
+the new functionality:
+
+- `apis/config/v1beta1/defaults_test.go`
+  - updated existing tests with `defaultObjectRetentionPolicies` which sets 
+  `FinishedWorkloadRetention` to nil which ensures that the default 
+  scenario maintains backward compatibility and no retention settings are set.
+  - new unit test: `set object retention policy for finished workloads` which
+  checks if provided can configuration overrides default values.
+- `pkg/config/config_test.go`
+  - updated existing tests with `defaultObjectRetentionPolicies` set to nil 
+  which tests if for existing tests the default behavior maintains backward
+  compatibility and results in no retention settings.
+  - new unit test: `objectRetentionPolicies config` which sets some example 
+  retention policy and then checks if it can be retrieved.
+- `pkg/controller/core/workload_controller_test.go`
+  - added following new tests:
+    - `shouldn't delete the workload because, object retention not configured`
+    - `shouldn't try to delete the workload (no event emitted) because it is already being deleted by kubernetes, object retention configured`
+    - `shouldn't try to delete the workload because the retention period hasn't elapsed yet, object retention configured`
+    - `should delete the workload because the retention period has elapsed, object retention configured`
+    - `should delete the workload immediately after it completed because the retention period is set to 0s`
+
+#### Integration tests
+
+<!--
+Describe what tests will be added to ensure proper quality of the enhancement.
+
+After the implementation PR is merged, add the names of the tests here.
+-->
+
+- Added behavior `Workload controller with resource retention` 
+in `test/integration/controller/core/workload_controller_test.go`
+
+
+[//]: # (### Graduation Criteria)
+
+[//]: # ()
+[//]: # (<!--)
+
+[//]: # ()
+[//]: # (Clearly define what it means for the feature to be implemented and)
+
+[//]: # (considered stable.)
+
+[//]: # ()
+[//]: # (If the feature you are introducing has high complexity, consider adding graduation)
+
+[//]: # (milestones with these graduation criteria:)
+
+[//]: # (- [Maturity levels &#40;`alpha`, `beta`, `stable`&#41;][maturity-levels])
+
+[//]: # (- [Feature gate][feature gate] lifecycle)
+
+[//]: # (- [Deprecation policy][deprecation-policy])
+
+[//]: # ()
+[//]: # ([feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md)
+
+[//]: # ([maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions)
+
+[//]: # ([deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+[//]: # (-->)
```

Comment:
Could you revert these comment-outs since we need to revisit here when we want to graduate this feature to Beta/Stable?

Reply:
I explicitly commented it out instead of removing it so it could be revisited later but maybe it needs to be there from the beginning? It's my first KEP ever so I'm learning as I go and I'll follow your guidance 🙏